tests/test_speed.py

```diff
@@ -0,0 +1,66 @@
+from swsscommon import swsscommon
+import time
+import re
+import json
+import os
+
+class TestSpeedSet(object):
+    num_ports = 32
+    def test_SpeedAndBufferSet(self, dvs):
+        speed_list = ['50000', '25000', '40000', '10000', '100000']
+
+        cdb = swsscommon.DBConnector(4, dvs.redis_sock, 0)
+        adb = swsscommon.DBConnector(1, dvs.redis_sock, 0)
+        cfg_port_table = swsscommon.Table(cdb, "PORT", '|')
+        cfg_buffer_profile_table = swsscommon.Table(cdb, "BUFFER_PROFILE", '|')
+        cfg_buffer_pg_table = swsscommon.Table(cdb, "BUFFER_PG", '|')
+        asic_port_table = swsscommon.Table(adb, "ASIC_STATE:SAI_OBJECT_TYPE_PORT")
+        asic_profile_table = swsscommon.Table(adb, "ASIC_STATE:SAI_OBJECT_TYPE_BUFFER_PROFILE")
+
+        buffer_profiles = cfg_buffer_profile_table.getKeys()
+        expected_buffer_profiles_num = len(buffer_profiles)
+        assert expected_buffer_profiles_num == 7
+        assert len(asic_profile_table.getKeys()) == expected_buffer_profiles_num
+
+        for speed in speed_list:
+            fvs = swsscommon.FieldValuePairs([("speed", speed)])
+            # set same speed on all ports
+            for i in range(0, self.num_ports):
+                cfg_port_table.set("Ethernet%d" % (i*4), fvs)
```

Comment:
I suspect this change won't work for all devices.

It all depends on what is self.num_ports?

For instance, Arista7260CX3 has 64 physical ports, if self.num_ports is 64 for this device, then it is fine. However, when Arista7260CX3 is used for T0, it will have 56 out of 64 ports break into 2 50G ports, bring the total number of ports to 112 + 8 = 120. Most of these ports are Etherenet(x * 4) and Ethernet(x * 4 + 2) except the 8 100G ports. So if self.num_ports is 120 for this device, then the code here is incorrect.

Reply:
It is designed to run on `PLATFORM=vs`

Reply:
I see. Can you at least add a comment at beginning of this file?

Thanks,
Ying